src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -6,20 +6,30 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class PegoutSignedCacheImpl implements PegoutSignedCache {
+public class PegoutSignedCacheImpl implements PegoutSignedCache {
 
   private static final Logger logger = LoggerFactory.getLogger(PegoutSignedCacheImpl.class);
+  private static final Integer CLEANUP_INTERVAL_IN_HOURS = 1;
 
   private final Map<Keccak256, Instant> cache = new ConcurrentHashMap<>();
+  private final ScheduledExecutorService cleanupScheduler = Executors.newSingleThreadScheduledExecutor();
   private final Duration ttl;
 
-  PegoutSignedCacheImpl(Duration ttl) {
-    assertValidTtl(ttl);
+  public PegoutSignedCacheImpl(Duration ttl) {
+    this.ttl = validateTtl(ttl);
```

Comment:
Not sure about this change, I think it looked better before.

It seems strange that a method called `validateTtl` returns a TTL value.

Reply:
Agree with @marcos-iov

Reply:
I also agree.

Reply:
Sounds good, I have updated to what I had before :)